src/Utility/MultipartFormDataBinaryContent.cs

```diff
@@ -120,27 +120,46 @@ public static void AddContentTypeHeader(HttpContent content, string contentType)
         content.Headers.ContentType = header;
     }
 
+#if NET
+    private static string CreateBoundary() =>
+        string.Create(BoundaryLength, 0, (chars, _) =>
+        {
+            Span<byte> random = stackalloc byte[BoundaryLength];
+            Random.Shared.NextBytes(random);
+
+            for (int i = 0; i < chars.Length; i++)
+            {
+                chars[i] = BoundaryValues[random[i] % BoundaryValues.Length];
```

Comment:
I expect `random[i] % BoundaryValues.Length` is equivalent to the operation using `Mask` below -- is there an advantage to doing it one way or the other?  If so, would it make sense to use the same approach in both places to make the code easier to understand?

Reply:
> I expect random[i] % BoundaryValues.Length is equivalent to the operation using Mask below

On .NET Core, yes. You can see here that the two methods produce identical assembly:
[SharpLab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4axAMxNSDIQwDeNBqYZiovBdgwwGkLrgwNePBgCEI3ACbYoATwAZGC4AcwwACwYAXgYAdjoNajNzS2tbewhHZ2ZGTx8/fwA1bEkOVRiGACI6clItFABWJDiADgBOaIBBdyEAEQBRADEAcQAJAEkAKQBpQIBZADkAeQAFAEU2WQAVAFUigHUADQBNAC0AfWxgMG8YADNQiN4AKwBrSXwuCDEARygnBwFAB3BD+ABeVSSKRMZm0TBQDHm5AAFLIxNguAAeMARPwAPnseIBAEpYaZjMkUmZ0ZiscB/LZCVBMd5RJUnNgwG9SpIIGAGAzbABtfJcXwBYJhSIAXWh1NMbFZoiGUkkKJZ4tEJPlCvu0AYKNczl4lUSLgYWKJflwLCl4QiahcAGpnWSqQrKQqFbibcLeDLKmKJcVSuVcMLNWz8P7AwBSDxecWFEplVR2kIOuXk6kAXxz+Y9DBz8OIiPmpDRGOxvqghNruHdKS93tp2KFMGZyvwHIwXJ5kj5Ao7oqTIftst11KVWvwqsHGu7OpzKX1UENxotsXNpqtDYz0sdLrdK7MLe9ZgbsaDY5TYdUke714AZAwkFps0WUoXvzmc8L5hgSIIG8CZxHVQDgNA8DljEPgsltRYIAmLhJFcVxQhJGUcwsKwbDsZgkARBgZ2jed1TbLF3EZTsGFwasSSMQtcyAA===)

That's unfortunately not the case with an older JIT as exists in .NET Framework, e.g. note the idiv on the left here:
[SharpLab](https://sharplab.io/#v2:EYLgHgbALANALiAhgZwLYB8ACAmAjAWAChMAGAAk1wDoAlAVwDs4BLVAUyoGEB7VAB2YAbNgCcAyqIBuzAMZtkAbiKYAzBWxlOZAN5Ey+snxHNJiOGzIzuDZHDLMmZAELdGAE0QiAngBk2DAHM4AAsyAF4yAHYSJUIDQ2NTc0trWwpcchd3Ty8ANURBOnlwsgAiElxsFSgAVghIgA4ATjCAQSdOABEAUQAxAHEACQBJACkAaR8AWQA5AHkABQBFGjEAFQBVXIB1AA0ATQAtAH1EYBk3NgAzAODmACsAa0FUBm4+AEcRWzpJAHcwF4AF6lWLxPQGVQUKBkKa4AAUYj4iAYAB4ZMFPAA+SyY74ASgh+l0cXiBiRKNRwC85hxIhRbl4JVsiBkjwKgm4MjI1PMAG0sgwPN4/IEQgBdMFk/Q0Bm8XpCQTw+lC3j4qXSq7cERkeEOOzMEoxexkVG4zzIKiioLBBT2ADU9sJpOlJOl0oxFr5zHFJUFwryBSKyD5KsZqG9voApM5XEKcvlCvIrf4bZKiWSAL4Z7MusgZqGYGFTbCI5Foz0iHGV5DO+Ju90UtG8th0uWoZlwVnswSc7ktgVxgPWiUasmy1WoBW95Xt9UZ+JanV6xyGiLGw1mmspsW2h1OhcGBvugw1yN+ocJoPyUPt88AMjIEBU6bz8Vz74zGb5UzYIW4bjDPwSq/v+gHAXMfAsKkVAzNwwwMIIDgOAE+LihmRgmGYFiUBA0JkBO4bTkqTaok4NKtmQyDlviOi5pmQA==)

I could make them both `% BoundaryValues.Length` if we're not concerned about the (arguably inconsequential) overhead on netfx

Reply:
Ah, no, very cool.  If you wouldn't mind adding a comment to that effect, I'm happy to keep the optimization.  Thanks!